pkg/config/configobserver/configobserver.go

```diff
@@ -262,64 +104,89 @@ func (c *Controller) ConfigChanged() (configCh <-chan struct{}, closeFn func())
 	}
 }
 
+// Fetches the token from cluster secret key
+func (c *Controller) fetchSecret(ctx context.Context, key string) (*v1.Secret, error) {
+	secret, err := c.kubeClient.CoreV1().Secrets("openshift-config").Get(ctx, key, metav1.GetOptions{})
+	if err != nil {
+		if errors.IsNotFound(err) {
+			klog.V(4).Infof("%s secret does not exist", key)
+			err = nil
+		} else if errors.IsForbidden(err) {
+			klog.V(2).Infof("Operator does not have permission to check %s: %v", key, err)
+			err = nil
+		} else {
+			err = fmt.Errorf("could not check %s: %v", key, err)
+		}
+	}
+
+	return secret, err
+}
+
+// Updates the stored tokens from the secrets in the cluster. (if present)
+func (c *Controller) updateToken(ctx context.Context) error {
+	klog.V(2).Infof("Refreshing configuration from cluster pull secret")
+	secret, err := c.fetchSecret(ctx, "pull-secret")
+	if err != nil {
+		return err
+	}
+
+	var nextConfig config.Controller
+	if secret != nil {
+		var token string
+		token, err = tokenFromSecret(secret)
```

Comment:
nit: It would be simpler to go with `token, err := tokenFromSecret(secret)` and save one line :)

Reply:
In this case, I'll be shadowing the `err` at L#130. I can give it a try, but almost sure that the lining will point the shadowing.

Reply:
Ahh ok np then...